boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,346 @@
+}
+
+void Io_Efuse_DisableChannel0(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(
+        efuse->channel0_port, efuse->channel0_pin, GPIO_PIN_RESET);
+}
+
+void Io_Efuse_EnableChannel1(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(efuse->channel1_port, efuse->channel1_pin, GPIO_PIN_SET);
+}
+
+void Io_Efuse_DisableChannel1(const struct Efuse_Context *const efuse)
+{
+    HAL_GPIO_WritePin(
+        efuse->channel1_port, efuse->channel1_pin, GPIO_PIN_RESET);
+}
+
+ExitCode Io_Efuse_ConfigureChannelMonitoring(
+    uint8_t                     monitoring_function,
+    struct Efuse_Context *const efuse)
+{
+    uint16_t register_value;
+
+    // Read original content of GCR Register
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_ReadRegister(SI_GCR_ADDR, &register_value, efuse));
+
+    CLEAR_BIT(register_value, (CSNS1_EN_MASK | CSNS0_EN_MASK));
+    SET_BIT(
+        register_value,
+        (monitoring_function & (CSNS1_EN_MASK | CSNS0_EN_MASK)));
+
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_WriteRegister(SI_GCR_ADDR, register_value, efuse));
+
+    return EXIT_CODE_OK;
+}
+
+ExitCode Io_Efuse_ExitFailSafeMode(struct Efuse_Context *const efuse)
+{
+    // Set the WDIN bit (15th bit) to exit out of fail-safe mode, the parity bit
+    // (14th bit) will automatically be set by the Io_Efuse_WriteRegister()
+    // function 1_1_00000_00000_0000
+    efuse->wdin_bit_to_set = true;
+
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_WriteRegister(SI_STATR_0_ADDR, 0x0000, efuse));
+
+    // Disable the watchdog timer
+    RETURN_CODE_IF_EXIT_NOT_OK(
+        Io_Efuse_WriteRegister(SI_GCR_ADDR, GCR_CONFIG, efuse));
+
+    // Check if the the efuse is still in fail-safe mode
+    if (Io_Efuse_IsEfuseInFailSafeMode(efuse))
+    {
+        return EXIT_CODE_UNIMPLEMENTED;
+    }
+
+    return EXIT_CODE_OK;
+}
+
+bool Io_Efuse_IsEfuseInFaultMode(const struct Efuse_Context *const efuse)
+{
+    return HAL_GPIO_ReadPin(efuse->fsb_port, efuse->fsb_pin);
+}
+
+bool Io_Efuse_IsEfuseInFailSafeMode(const struct Efuse_Context *const efuse)
+{
+    return HAL_GPIO_ReadPin(efuse->fsob_port, efuse->fsob_pin);
+}
+
+void Io_Efuse_DelatchFaults(const struct Efuse_Context *const efuse)
+{
+    // Delatch the latchable faults by alternating channel 0's and channel 1's
+    // input pins high->low->high.
+    HAL_GPIO_WritePin(efuse->channel0_port, efuse->channel0_pin, GPIO_PIN_SET);
+    HAL_GPIO_WritePin(
+        efuse->channel0_port, efuse->channel0_pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(efuse->channel0_port, efuse->channel0_pin, GPIO_PIN_SET);
+    HAL_GPIO_WritePin(efuse->channel1_port, efuse->channel1_pin, GPIO_PIN_SET);
+    HAL_GPIO_WritePin(
+        efuse->channel1_port, efuse->channel1_pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(efuse->channel1_port, efuse->channel1_pin, GPIO_PIN_SET);
+}
+
+ExitCode Io_Efuse_WriteRegister(
+    uint8_t                     register_address,
```

Comment:
I think it would be nice to indicate these as tx_register_address and tx_register_value. rx for the function below.